homeassistant/components/modbus/climate.py

```diff
@@ -37,158 +45,131 @@
+        self._slave = config[CONF_SLAVE]
+        self._target_temperature_register = config[CONF_TARGET_TEMP]
+        self._current_temperature_register = config[CONF_CURRENT_TEMP]
+        self._current_temperature_register_type = config[
+            CONF_CURRENT_TEMP_REGISTER_TYPE
+        ]
         self._target_temperature = None
         self._current_temperature = None
-        self._data_type = data_type
-        self._count = int(count)
-        self._precision = precision
-        self._scale = scale
-        self._offset = offset
-        self._unit = unit
-        self._max_temp = max_temp
-        self._min_temp = min_temp
-        self._temp_step = temp_step
-        self._structure = ">f"
+        self._data_type = config[CONF_DATA_TYPE]
+        self._structure = config[CONF_STRUCTURE]
+        self._count = config[CONF_DATA_COUNT]
+        self._precision = config[CONF_PRECISION]
+        self._scale = config[CONF_SCALE]
+        self._scan_interval = timedelta(seconds=config[CONF_SCAN_INTERVAL])
+        self._offset = config[CONF_OFFSET]
+        self._unit = config[CONF_UNIT]
+        self._max_temp = config[CONF_MAX_TEMP]
+        self._min_temp = config[CONF_MIN_TEMP]
+        self._temp_step = config[CONF_STEP]
+        self._hvac_mode = HVAC_MODE_AUTO
         self._available = True
 
-        data_types = {
-            DATA_TYPE_INT: {1: "h", 2: "i", 4: "q"},
-            DATA_TYPE_UINT: {1: "H", 2: "I", 4: "Q"},
-            DATA_TYPE_FLOAT: {1: "e", 2: "f", 4: "d"},
-        }
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        async_track_time_interval(
+            self.hass, lambda arg: self._update(), self._scan_interval
+        )
 
-        self._structure = f">{data_types[self._data_type][self._count]}"
+    @property
+    def should_poll(self):
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+
+        # Handle polling directly in this entity
+        return False
 
     @property
     def supported_features(self):
         """Return the list of supported features."""
         return SUPPORT_TARGET_TEMPERATURE
 
-    def update(self):
-        """Update Target & Current Temperature."""
-        self._target_temperature = self._read_register(
-            CALL_TYPE_REGISTER_HOLDING, self._target_temperature_register
-        )
-        self._current_temperature = self._read_register(
-            self._current_temperature_register_type, self._current_temperature_register
-        )
-
     @property
     def hvac_mode(self):
         """Return the current HVAC mode."""
-        return HVAC_MODE_AUTO
+        return self._hvac_mode
 
     @property
     def hvac_modes(self):
         """Return the possible HVAC modes."""
         return [HVAC_MODE_AUTO]
 
+    def set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        self._hvac_mode = hvac_mode
```

Comment:
This isn't used. Why do we implement this method?

Reply:
Apparently, Home Assistant expects this method to be implemented. I noticed exceptions in a log, because Home Assistant was trying to call this method, which was not implemented.

While we support only AUTO HVAC mode, I think it is safe to keep the method.

Reply:
What was calling this service?

Reply:
![Snímka obrazovky z 2020-10-01 09-20-17](https://user-images.githubusercontent.com/4460766/94780616-090ba500-03c9-11eb-9eb7-dc00f2225713.png)

See the attached image. I think it is a button to change HVAC modes. In this case it calls the `set_hvac_mode()` method, but it will always pass the mode `AUTO`. This mode is the only one supported by the Modbus climate entity.

Reply:
Ok, we can simplify here and just define the method without any function body.

Reply:
OK, I'll update the code.

Reply:
Code updated.